In [130]:
import pandas as pd
import geopandas as gpd

In [131]:
school_external = pd.read_csv('../data/landing/school_location.csv', encoding='latin-1')


In [132]:
school_external.head()

Education_Sector  Entity_Type  School_No  \
0       Government            1          1   
1       Government            1          3   
2       Government            1          4   
3       Government            1          8   
4       Government            1         12   

                              School_Name School_Type School_Status  \
0                 Alberton Primary School     Primary             O   
1  Allansford and District Primary School     Primary             O   
2                    Avoca Primary School     Primary             O   
3                   Avenel Primary School     Primary             O   
4               Warrandyte Primary School     Primary             O   

       Address_Line_1 Address_Line_2 Address_Town Address_State  ...  \
0   21 Thomson Street            NaN     Alberton           VIC  ...   
1        Frank Street            NaN   Allansford           VIC  ...   
2  118 Barnett Street            NaN        Avoca           VIC  ...   
3  40 Anderson Street            NaN       Avenel           VIC  ...   
4  5-11 Forbes Street            NaN   Warrandyte           VIC  ...   

   Postal_Address_Line_1 Postal_Address_Line_2 Postal_Town Postal_State  \
0      21 Thomson Street                   NaN    ALBERTON          VIC   
1           Frank Street                   NaN  ALLANSFORD          VIC   
2             P O Box 12                   NaN       AVOCA          VIC   
3     40 Anderson Street                   NaN      AVENEL          VIC   
4     5-11 Forbes Street                   NaN  WARRANDYTE          VIC   

  Postal_Postcode  Full_Phone_No LGA_ID         LGA_Name          X         Y  
0            3971   03 5183 2412    681   Wellington (S)  146.66660 -38.61771  
1            3277   03 5565 1382    673  Warrnambool (C)  142.59039 -38.38628  
2            3467   03 5465 3176    599     Pyrenees (S)  143.47565 -37.08450  
3            3664   03 5796 2264    643  Strathbogie (S)  145.23472 -36.90137  
4            3113   03 9844 3537    421   Manningham (C)  145.21398 -37.74268  

[5 rows x 21 columns]

In [133]:
school_external.columns

Index(['Education_Sector', 'Entity_Type', 'School_No', 'School_Name',
       'School_Type', 'School_Status', 'Address_Line_1', 'Address_Line_2',
       'Address_Town', 'Address_State', 'Address_Postcode',
       'Postal_Address_Line_1', 'Postal_Address_Line_2', 'Postal_Town',
       'Postal_State', 'Postal_Postcode', 'Full_Phone_No', 'LGA_ID',
       'LGA_Name', 'X', 'Y'],
      dtype='object')

In [134]:
school_external = school_external[['School_Name','School_Type', 'Postal_Town', 'X', 'Y']]

In [135]:
school_external.drop_duplicates()

School_Name School_Type  \
0                          Alberton Primary School     Primary   
1           Allansford and District Primary School     Primary   
2                             Avoca Primary School     Primary   
3                            Avenel Primary School     Primary   
4                        Warrandyte Primary School     Primary   
...                                            ...         ...   
2297                          Plenty River College   Secondary   
2298            Holy Cross Catholic Primary School     Primary   
2299                         Sidrah Gardens School     Primary   
2300           Mountain District Community College   Secondary   
2301  Mary Queen of Heaven Catholic Primary School     Primary   

             Postal_Town          X         Y  
0               ALBERTON  146.66660 -38.61771  
1             ALLANSFORD  142.59039 -38.38628  
2                  AVOCA  143.47565 -37.08450  
3                 AVENEL  145.23472 -36.90137  
4             WARRANDYTE  145.21398 -37.74268  
...                  ...        ...       ...  
2297        SOUTH MORANG  145.08148 -37.64875  
2298           MICKLEHAM  144.90520 -37.53046  
2299  NARRE WARREN NORTH  145.31589 -37.97324  
2300      FERNTREE GULLY  145.29327 -37.88319  
2301           GREENVALE  144.88566 -37.62974  

[2302 rows x 5 columns]

In [136]:
school_external['School_Type'].unique()

array(['Primary', 'Pri/Sec', 'Special', 'Camp', 'Secondary', 'Language'],
      dtype=object)

In [137]:
school_external = school_external[school_external['School_Type'].isin(['Primary', 'Pri/Sec', 'Secondary', 'Language'])]

In [138]:
len(school_external)

2182

### Now there are 2182 schools, including primary, secondary and language schools
### Next, we consider adding university locations

In [139]:
file_path = '../data/landing/Uni & College.geojson'
university_data = gpd.read_file(file_path)


In [140]:
university_data.columns

Index(['id', '@id', 'addr:city', 'addr:housenumber', 'addr:postcode',
       'addr:state', 'addr:street', 'addr:suburb', 'alt_name', 'amenity',
       'branch', 'check_date', 'contact:email', 'contact:facebook',
       'contact:phone', 'description', 'email', 'gnis:feature_id',
       'internet_access', 'internet_access:fee', 'is_in', 'mobile', 'name',
       'name:en', 'note', 'opening_hours', 'operator', 'operator:short',
       'operator:type', 'phone', 'source', 'website', 'wheelchair', 'wikidata',
       'wikipedia', 'geometry'],
      dtype='object')

In [141]:
university_data = university_data[['amenity', 'name', 'geometry']]

In [142]:
university_data['X'] = university_data['geometry'].x
university_data['Y'] = university_data['geometry'].y
university_data['School_Type'] = university_data['amenity']
university_data['School_Name'] = university_data['name']
university_data = university_data.drop(columns='amenity').drop(columns='name')


### There are 75 instances about university and college

In [143]:
university_data

geometry           X          Y School_Type  \
0   POINT (145.36439 -38.08443)  145.364387 -38.084432     college   
1    POINT (-96.98145 28.81556)  -96.981454  28.815559     college   
2   POINT (143.85879 -37.56115)  143.858787 -37.561148     college   
3   POINT (144.97215 -37.81031)  144.972154 -37.810313  university   
4   POINT (145.14910 -37.87686)  145.149097 -37.876861     college   
..                          ...         ...        ...         ...   
70  POINT (144.96086 -37.80755)  144.960860 -37.807551     college   
71  POINT (144.93912 -37.79757)  144.939123 -37.797570     college   
72  POINT (145.28310 -37.80113)  145.283103 -37.801126     college   
73  POINT (144.96433 -37.80004)  144.964334 -37.800037  university   
74  POINT (144.96810 -37.82469)  144.968104 -37.824694  university   

                                      School_Name  
0   Hillcrest Christian College (Ayr Hill Campus)  
1                                Victoria College  
2                                    Arts Academy  
3                           James Cook University  
4                 Huanya Cultural Training Centre  
..                                            ...  
70                    Australian National College  
71                   Barkly International College  
72                                    U3A Croydon  
73                          Telstra Creator Space  
74                                   Wilin Centre  

[75 rows x 5 columns]

In [144]:
school_combined = pd.concat([university_data, school_external], axis=0)

### In total, there are 2257 instances about school (Secondary, Primary, Language, University, College) locations

In [146]:
school_combined.drop_duplicates()

geometry           X          Y School_Type  \
0     POINT (145.36439 -38.08443)  145.364387 -38.084432     college   
1      POINT (-96.98145 28.81556)  -96.981454  28.815559     college   
2     POINT (143.85879 -37.56115)  143.858787 -37.561148     college   
3     POINT (144.97215 -37.81031)  144.972154 -37.810313  university   
4     POINT (145.14910 -37.87686)  145.149097 -37.876861     college   
...                           ...         ...        ...         ...   
2297                         None  145.081480 -37.648750   Secondary   
2298                         None  144.905200 -37.530460     Primary   
2299                         None  145.315890 -37.973240     Primary   
2300                         None  145.293270 -37.883190   Secondary   
2301                         None  144.885660 -37.629740     Primary   

                                        School_Name         Postal_Town  
0     Hillcrest Christian College (Ayr Hill Campus)                 NaN  
1                                  Victoria College                 NaN  
2                                      Arts Academy                 NaN  
3                             James Cook University                 NaN  
4                   Huanya Cultural Training Centre                 NaN  
...                                             ...                 ...  
2297                           Plenty River College        SOUTH MORANG  
2298             Holy Cross Catholic Primary School           MICKLEHAM  
2299                          Sidrah Gardens School  NARRE WARREN NORTH  
2300            Mountain District Community College      FERNTREE GULLY  
2301   Mary Queen of Heaven Catholic Primary School           GREENVALE  

[2257 rows x 6 columns]

In [147]:
school_combined.to_csv('../data/raw/school_location.csv', index=False) 